src/DataStructures/Tensor/Expressions/AddSubtract.hpp

```diff
@@ -268,12 +334,88 @@ struct AddSub<T1, T2, ArgsList1<Args1...>, ArgsList2<Args2...>, Sign>
   SPECTRE_ALWAYS_INLINE decltype(auto) get(
       const std::array<size_t, num_tensor_indices>& result_multi_index) const {
     if constexpr (std::is_same_v<tmpl::list<Args1...>, tmpl::list<Args2...>>) {
-      return add_or_subtract(result_multi_index, result_multi_index);
+      if constexpr (op1_spatial_spacetime_index_positions.size() != 0 or
+                    op2_spatial_spacetime_index_positions.size() != 0) {
+        // Operands have the same generic index order, but at least one of them
+        // has at least one spacetime index where a spatial index has been used,
+        // so we need to compute the 2nd operand's (possibly) shifted
+        // multi-index values
```

Comment:
after thinking about it enough I was able to figure out why (I think) only op2 needs to be changed, but maybe add a short sentence about that?

Reply:
Yes, will do. In short, the reason is that the first operand's generic index order is the index order used for the `AddSub` expression (i.e. the index order of the sum/result tensor), so the first operand's index order is what is "propagated" up via its generic index order being what is inserted into the tree